libs/ledger-live-common/src/api/Ethereum.ts

```diff
@@ -147,6 +153,22 @@ export type API = {
   }>;
   getBlockByHash: (blockHash: string) => Promise<BlockByHashOutput | undefined>;
 };
+
+function adaptTx(apiTx): Tx {
+  return {
+    ...apiTx,
+    value: new BigNumber(apiTx.value),
+    gas: new BigNumber(apiTx.gas),
+    gas_price: new BigNumber(apiTx.gas_price),
+    cumulative_gas_used: new BigNumber(apiTx.cumulative_gas_used),
+    gas_used: new BigNumber(apiTx.gas_used),
+    block: {
+      ...apiTx.block,
+      height: new BigNumber(apiTx.block.height),
```

Comment:
i would prefer we don't have to "patch" at this "api" level, to me the idea of `api` is it stays true to what the API return, and instead we need to parse whatever we need on the implementation level.

because typically here, we never will need block.height to be a BigNumber.

Reply:
Should be done